### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by mcalhoun**
## what
* Initial AWS Inspector implementation

## why
* Schedule AWS Inspector to run vulnerability and compliance checks against EC2 instances on a periodic basis

### CODE REVIEW

File: examples/complete/fixtures.us-east-2.tfvars [hunk 1]
```diff
@@ -1,10 +1,6 @@
...
-stage = "test"
-
-name = "example"
-
+region          = "us-east-2"
+namespace       = "eg"
+environment     = "ue2"
+stage           = "test"
+create_iam_role = true
+enabled_rules   = ["arn:aws:inspector:us-east-2:646659390643:rulespackage/0-m8r61nnh"]
```
Comment (osterman) [inline examples/complete/fixtures.us-east-2.tfvars:6]: Hrmmm.. tests should be able to run on other accounts (e.g. by contributors)

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,98 @@
...
+  attributes = concat(module.this.attributes, ["inspector", "assessment", "template"])
+  context    = module.this.context
+}
+
+resource "aws_inspector_assessment_template" "assessment" {
+  name               = module.insepector_assessment_template_label.id
+  target_arn         = aws_inspector_assessment_target.target.arn
+  duration           = var.assessment_duration
+  rules_package_arns = var.enabled_rules
+}
```
Comment (osterman) [inline main.tf:29]: Seems like there's enough meat that warrants a module for encapsulation?

Comment (osterman) [reply main.tf:29]: (perhaps including the stuff below as well)

Comment (osterman) [inline main.tf]: (ignore my earlier comments, I thought it was `examples/complete/main.tf`)

File: examples/complete/fixtures.us-east-2.tfvars (hunk 1 above)
Comment (mcalhoun) [reply examples/complete/fixtures.us-east-2.tfvars:6]: This is actually an ARN of a rules package from an AWS account, not locking it into on of our accounts.

See the [list of rules packages](https://docs.aws.amazon.com/inspector/latest/userguide/inspector_rules-arns.html) for possible ARNs. This is also documented in the `variables.tf` file.

File: main.tf [hunk 3]
```diff
@@ -1,13 +1,98 @@
...
+  version = "0.21.0"
+
+  attributes = concat(module.this.attributes, ["inspector", "schedule"])
+  context    = module.this.context
+}
+
+resource "aws_cloudwatch_event_rule" "schedule" {
+  count               = local.create_scheduled_event ? 1 : 0
+  name                = module.insepector_schedule_label.id
+  description         = "Trigger an AWS Inspector Assessment"
```
Comment (osterman) [inline main.tf]: lets parameterize in a var, with this as the defautl

Comment (mcalhoun) [reply main.tf]: 7df408c6f4ebaffe167ee3d0f3a776d0dd644845

**Comment by mcalhoun** [comment]
@osterman this is ready for another review when you have a moment

File: README.md [hunk 4]
```diff
@@ -25,8 +25,8 @@
 
 -->
 
-This is `terraform-aws-inspector` project provides all the scaffolding for a typical well-built Cloud Posse module. It's a template repository you can
-use when creating new repositories.
+This module enables [AWS Inspector](https://aws.amazon.com/inspector/) in one region of one account and optionally 
+enables various rules packages.
```
Comment (osterman) [inline README.md]:
```suggestion
enables [various rules packages provided by AWS](https://docs.aws.amazon.com/inspector/latest/userguide/inspector_rules-arns.html).
```

File: README.md [hunk 5]
```diff
@@ -73,9 +73,11 @@ For automated tests of the complete example using [bats](https://github.com/bats
 (which tests and deploys the example on AWS), see [test](test).
 
 ```hcl
-module "example" {
+module "inspector" {
   source = "https://github.com/cloudposse/terraform-aws-inspector.git?ref=master"
-  example = "Hello world!"
+  
+  create_iam_role = true
+  enabled_rules   = ["arn:aws:inspector:us-east-2:646659390643:rulespackage/0-m8r61nnh"]
```
Comment (osterman) [inline README.md:108]: Please also add `https://docs.aws.amazon.com/inspector/latest/userguide/inspector_rules-arns.html` to the `references:` section

File: README.md (hunk 4 above)
Comment (mcalhoun) [reply README.md]: 6da6e8128fb5cb7aa5607848a050acd35da7d13d

File: README.md (hunk 5 above)
Comment (mcalhoun) [reply README.md:108]: 6da6e8128fb5cb7aa5607848a050acd35da7d13d

**Review by osterman — APPROVED** [review]

**Pull request merged**